src/custom-elements/src/components/date-picker/date-picker.tsx

```diff
@@ -0,0 +1,145 @@
+import { Component, Element, Listen, Method, Prop } from '@stencil/core';
+import { contains, uuid4 } from '../../utils/utils';
+import { ESCAPE_KEYCODE } from '../../constants/constants';
+
+@Component({
+  tag: 'chi-date-picker',
+  scoped: true
+})
+export class DatePicker {
+  /**
+   * Selected date in the date picker
+   */
+  @Prop({ reflectToAttr: true, mutable: true }) value: string;
+
+  /**
+   * Locale to use in date picker
+   */
+  @Prop({ reflectToAttr: true }) locale = 'en';
+
+  /**
+   * Minimum eligible date
+   */
+  @Prop({ reflectToAttr: true }) min = '01/01/1900';
+
+  /**
+   * Maximum eligible date
+   */
+  @Prop({ reflectToAttr: true }) max = '12/31/2099';
+
+  /**
+   * Date format used in the attributes and how it will be shown to the user.
+   */
+  @Prop({ reflectToAttr: true }) format = 'MM/DD/YYYY';
+
+  /**
+   * Indicates whether the dropdown calendar is open or closed
+   */
+  @Prop({ reflectToAttr: true, mutable: true }) active = false;
+
+  @Element() el: HTMLElement;
+
+  private _input: HTMLInputElement;
+  private _uuid: string;
```

Comment:
@dani-cl-madrid can't we use State() for both _input and _uuid?

Reply:
@srinath-koneru  We could use is in _uuid but not in _input. Anyway, it doesn't make sense, as we don't want to rerender the component when this values change.